tests/post/steps/test_monitoring.py

```diff
@@ -179,4 +192,63 @@ def _node_has_metrics():
     utils.retry(_node_has_metrics, times=60, wait=3)
 
 
+@then(
+    "the deployed Prometheus alert rules are the same as the default "
+    "alert rules")
+def check_deployed_rules(host, prometheus_api):
+    def _wait_prom_rule_cmp():
+        try:
+            deployed_rules = prometheus_api.get_rules()
+        except utils.PrometheusApiError as exc:
+            pytest.fail(str(exc))
+
+        if not deployed_rules:
+            pytest.fail(
+                "Expected default deployed Prometheus rules but got {}"
+                .format(deployed_rules)
+            )
+
+        rule_group = deployed_rules.get('data', {}).get('groups', [])
+        deployed_alert_rules = []
+        for item in rule_group:
+            for rule in item.get('rules', []):
+                # rule type can be alerting or recording
+                # For now, we only need alerting rules
+                if rule['type'] == "alerting":
+                    message = rule['annotations'].get('message') or \
+                        rule['annotations'].get('summary')
+                    fixup_alerting_rule = {
+                        'name': rule['name'],
+                        'severity': rule['labels']['severity'],
+                        'message': message,
+                        'query': rule['query']
+                    }
+                    deployed_alert_rules.append(fixup_alerting_rule)
+
+        try:
+            with open(ALERT_RULE_FILE_PATH) as f:
+                try:
+                    default_alert_rules = json.load(f)
+                except json.JSONDecodeError as exc:
+                    pytest.fail("Failed to decode JSON from {}: {!s}".format(
+                        ALERT_RULE_FILE_PATH, exc)
+                    )
+        except IOError as exc:
+            pytest.fail(
+                "Failed to open file {}: {!s}"
+                .format(ALERT_RULE_FILE_NAME, exc)
+            )
+
+        assert default_alert_rules == deployed_alert_rules, (
+            "Expected default Prometheus rules to be equal to deployed rules."
+        )
+
+    utils.retry(
+        _wait_prom_rule_cmp,
+        times=5,
+        wait=3,
+        name="checking deployed Prometheus rules are equal to default rules"
+    )
```

Comment:
Not sure we need to implement a retry here since we do not change anything, just reading the API, the PrometheusAPI wrapper will take care of retrying in case of non 200 exit code.